lib/bundler/cli/config.rb

```diff
@@ -59,7 +59,7 @@ def run
               "This will take precedence over the global value you are setting"
           end
 
-          if global = locations[:global]
+          if global = locations[:global] and global != locations[:global]
```

Comment:
I think it's [Ruby style guide](https://github.com/bbatsov/ruby-style-guide) from bbathos